curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md

```diff
@@ -0,0 +1,69 @@
+---
+id: 60a3e3396c7b40068ad69971
+title: Part 8
+challengeType: 0
+dashedName: part-8
+---
+
+# --description--
+
+Time for CSS.
+
+Even though your `<div>` has no text, it's still treated as a box with content.
+Write a CSS rule that uses the `.canvas` class selector and set its `width` to 500 pixels.
+Here's a CSS rule that sets the width of the class `card` to 300 pixels:
+
+```css
+.card {
+  width: 300px;
+}
+```
+
+# --hints--
+
+Your code should have a `.canvas` selector.
+
+```js
+const hasCanvas = new __helpers.CSSHelp(document).getStyleDeclaration('.canvas');
+assert(hasCanvas)
+```
+
+You should set the `width` property to `500px`.
+
+```js
+const hasWidth = new __helpers.CSSHelp(document).getCSSRules().some(x => x.style.width === '500px')
+assert(hasWidth);
+```
+
+Your `.canvas` element should have a `width` of `500px`.
+
+```js
+const width = new __helpers.CSSHelp(document).getStyleDeclaration('.canvas').getPropertyValue('width');
```

Comment:
```js
frame-runner.js:98 TypeError: Cannot read property 'getPropertyValue' of undefined
    at eval (eval at <anonymous> (VM33234 frame-runner.40c5018811bdc1204b53.js:2), <anonymous>:1:77)
    at HTMLDocument.<anonymous> (VM33234 frame-runner.40c5018811bdc1204b53.js:2)
    at u (VM33234 frame-runner.40c5018811bdc1204b53.js:2)
    at c (VM33234 frame-runner.40c5018811bdc1204b53.js:2)
```
This gets explicit error.

Reply:
That error should only throw when the previous test fails, where the camper hasn't added the `.canvas` selector yet.

Reply:
Hmm - could maybe avoid the error with optional chaining. Will play with that.

Reply:
> avoid the error with optional chaining.

That is largely what I have been doing.

Reply:
Yeah, sorry, this was when I didn't have yet https://github.com/freeCodeCamp/freeCodeCamp/pull/42266 applied locally.